homeassistant/components/rointe/config_flow.py

```diff
@@ -0,0 +1,141 @@
+"""Config flow for Rointe Heaters integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from rointesdk.rointe_api import RointeAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CONF_INSTALLATION,
+    CONF_LOCAL_ID,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    DOMAIN,
+    LOGGER,
+)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for Rointe Heaters."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Config flow init."""
+        self.step_user_data: dict[str, Any] | None = None
+        self.step_user_local_id: str | None = None
+        self.step_user_installations: dict[str, Any] | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Step: User credentials validation."""
+
+        errors: dict[str, str] = {}
+        rointe_api: RointeAPI
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            )
+
+        rointe_api = RointeAPI(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+
+        login_error_code = await self.hass.async_add_executor_job(
+            rointe_api.initialize_authentication
+        )
+
+        if not login_error_code.success or not rointe_api.is_logged_in():
+            return self.async_show_form(
+                step_id="user",
+                data_schema=STEP_USER_DATA_SCHEMA,
+                errors={"base": login_error_code.error_message},
```

Comment:
Reminder that every error message will need to be reflected in `strings.json` for translations—I don't currently see those. If you don't know every possible message (or don't want to manage so many strings), you might consider a generalized message for the UI and logging an error with the actual message.